File: python/src/starter.py
import asyncio
import uuid
from temporalio.client import Client
from temporalio.envconfig import ClientConfigProfile
import shared
import workflows


async def main():
    default_profile = ClientConfigProfile.load()
    connect_config = default_profile.to_client_connect_config()

    client = await Client.connect(**connect_config)
    result = await client.execute_workflow(
        workflows.HelloWorldWorkflow,
        "Hacker",
        id=f"hello-world-workflow-{uuid.uuid4()}",
        task_queue=shared.TASK_QUEUE_NAME,
    )
    print("Workflow result:", result)


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/src/starter.ts
import { Client, Connection } from '@temporalio/client';
import { nanoid } from 'nanoid';

import { TASK_QUEUE_NAME } from './shared';
import { helloWorldWorkflow } from './workflows';

async function run() {
  const connection = await Connection.connect({
    address: process.env.TEMPORAL_ADDRESS,
    tls: process.env.TEMPORAL_TLS === 'true',
    apiKey: process.env.TEMPORAL_API_KEY,
  });

  const client = new Client({
    connection,
    namespace: process.env.TEMPORAL_NAMESPACE,
  });

  const handle = await client.workflow.start(helloWorldWorkflow, {
    taskQueue: TASK_QUEUE_NAME,
    args: ['Hacker'],
    // in practice, use a meaningful business ID, like customerId or transactionId
    workflowId: 'hello-world-' + nanoid(),
  });
  console.log(`Started workflow ${handle.workflowId}`);

  // optional: wait for client result
  console.log(await handle.result());
}

run().catch((err) => {
  console.error(err);
  process.exit(1);
});
